pandas/tests/groupby/test_groupby.py

```diff
@@ -5777,6 +5769,81 @@ def test_agg_over_numpy_arrays(self):
 
         assert_frame_equal(result, expected)
 
+    def test_cummin_cummax(self):
+        # GH 15048
+        # Test with different dtypes
+        for dtype in [np.int32, np.float32, np.float64]:
+            df = pd.DataFrame({'A': [1, 1, 1, 2, 2, 2],
+                               'B': range(3, 9)}).astype(dtype)
+            result = df.groupby('A').cummin()
+            expected = pd.DataFrame({'B': [3, 3, 3, 6, 6, 6]}).astype(dtype)
+            tm.assert_frame_equal(result, expected)
+            expected = df.groupby('A').B.apply(lambda x: x.cummin()).to_frame()
+            tm.assert_frame_equal(result, expected)
+            # Test min value for dtype
+            try:
```

Comment:
instead of this, pass these in with the dtype, IOW change your loop to be

```
for dtype, min_value in [(np.int32, np.iinfo(np.int32).min), (np.float32.....)]
```